packages/rollup-plugin-import-meta-assets/src/rollup-plugin-import-meta-assets.js

```diff
@@ -62,19 +73,96 @@ function importMetaAssets({ include, exclude, warnOnError, transform } = {}) {
 
   return {
     name: 'rollup-plugin-import-meta-assets',
-
     async transform(code, id) {
       if (!filter(id)) {
         return null;
       }
 
-      const ast = this.parse(code);
-      const magicString = new MagicString(code);
+      // Part 1: resolve dynamic template literal expressions
+      const parsed = this.parse(code);
+
+      let dynamicURLIndex = -1;
+      let ms;
+
+      await asyncWalk(parsed, {
+        enter: async node => {
+          const importMetaUrlType = getImportMetaUrlType(node);
+
+          if (importMetaUrlType !== 'dynamic') {
+            return;
+          }
+          dynamicURLIndex += 1;
+
+          try {
+            // see if this is a Template Literal with expressions inside, and generate a glob expression
+            const glob = dynamicURLToGlob(node.arguments[0], code.substring(node.start, node.end));
+
+            if (!glob) {
+              // this was not a variable dynamic url
+              return;
+            }
+
+            const { globby } = await import('globby');
+            // execute the glob
+            const result = await globby(glob, { cwd: path.dirname(id) });
+            const paths = result.map(r =>
+              r.startsWith('./') || r.startsWith('../') ? r : `./${r}`,
+            );
+
+            // create magic string if it wasn't created already
+            ms = ms || new MagicString(code);
+            // unpack variable dynamic url into a function with url statements per file, rollup
+            // will turn these into chunks automatically
+            ms.prepend(
+              `function __variableDynamicURLRuntime${dynamicURLIndex}__(path) {
+  switch (path) {
+${paths.map(p => `    case '${p}': return new URL('${p}', import.meta.url);`).join('\n')}
+${`    default: return new Promise(function(resolve, reject) {
+      (typeof queueMicrotask === 'function' ? queueMicrotask : setTimeout)(
+        reject.bind(null, new Error("Unknown variable dynamic new URL statement: " + path))
+      );
+    })\n`}   }
+ }\n\n`,
+            );
+            // call the runtime function instead of doing a dynamic url, the url specifier will
+            // be evaluated at runtime and the correct url will be returned by the injected function
+            ms.overwrite(
+              node.start,
+              node.start + 7,
+              `__variableDynamicURLRuntime${dynamicURLIndex}__`,
+            );
+          } catch (error) {
+            if (error instanceof VariableURLError) {
+              if (warnOnError) {
+                this.warn(error);
+              } else {
+                this.error(error);
+              }
+            } else {
+              this.error(error);
+            }
```

Comment:
Could be simplified to this?

```suggestion
            if (error instanceof VariableURLError && warnOnError) {
                this.warn(error);
            } else {
              this.error(error);
            }
```

Reply:
Done